### PR TITLE
Customize Your Store - Footer

### LINKED ISSUE #39719: [Customize Your Store] Footer
## User Story

> As a new merchant, I would like a to fine tune the AI design, so I can add my personal touch.

## Dev

Should be similar to header implementation, please refer to:

- https://github.com/woocommerce/woocommerce/issues/39718

## Design

* [Figma](https://www.figma.com/file/Mrk6SERPZ4KrFHSjM0a8TK/Customize-your-store-with-pattern-assembler?node-id=2631%3A255291&mode=dev)

Image

## Acceptance Criteria

```
Scenario: Footer panel is opened
    Given the user opens the header panel
    Then available footer patterns are shown

Scenario: User changes the footer 
    Given the changes the footer
    Then the footer should be updated in the preview pane

Scenario: Save
    Given the user presses the save button
    Then the rendered preview should be saved permanently
```

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #39719  .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use WP 6.3 
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper and enable `customize-store` feature flag
4. Go to `WooCommerce > Home`
5. Click on `Customize your store` task 
6. Click on `Assembler Hub` button
7. Click on `Change your footer`
8. Click on one of the footer blocks from the sidebar.
9. Confirm the selected footer block renders on the preview.
 

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add `Change your footer` sidebar to customize your store task.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/hooks/use-pattern.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable @woocommerce/dependency-group */
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/hooks/use-pattern.ts]: There is a same hook:  [`plugins/woocommerce-admin/client/customize-store/assembler-hub/hooks/use-patterns.ts`](https://github.com/woocommerce/woocommerce/blob/378273d6270339ba5f9e07d1817f33fa909e99ca/plugins/woocommerce-admin/client/customize-store/assembler-hub/hooks/use-patterns.ts). Should we use it? 😅

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx [hunk 2]
```diff
@@ -2,21 +2,84 @@
...
+
+const footerPatternNames = [
+	'woocommerce-blocks/footer-large',
+	'woocommerce-blocks/footer-large-dark',
+	'woocommerce-blocks/footer-simple',
+	'woocommerce-blocks/footer-simple-dark',
+	'woocommerce-blocks/footer-simple-menu-and-cart',
+	'woocommerce-blocks/footer-with-2-menus',
+	'woocommerce-blocks/footer-with-2-menus-dark',
+	'woocommerce-blocks/footer-with-3-menus',
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx:31]: Could you please update this list to only include these? We only need to show them. 🙂 

- woocommerce-blocks/footer-large
- woocommerce-blocks/footer-simple-menu-and-cart	
- woocommerce-blocks/footer-with-3-menus

https://github.com/woocommerce/woocommerce/issues/39719#issuecomment-1718882324

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx [hunk 3]
```diff
@@ -2,21 +2,84 @@
...
+				attributes: {
+					...selectedBlocks[ 0 ].attributes,
+					slug: 'footer',
+				},
+			};
+
+			onChange(
+				[
+					...blocks.map( ( block ) => {
+						if ( block.attributes?.slug === 'footer' ) {
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx]: There is an issue (p1694583571487789-slack-C01SFMVEYAK) when using slug. We could replace the last block here. 😃

**Review by chihsuan — COMMENTED** [review]
Thanks for working on this quickly @moon0326! 💯 

Just left some minor comments.

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx (hunk 2 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx:31]: Updated in https://github.com/woocommerce/woocommerce/pull/40136/commits/53a6cd884eb3daf2c633aaff4cdff5bc959cf782

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/hooks/use-pattern.ts (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/hooks/use-pattern.ts]: Updated in https://github.com/woocommerce/woocommerce/pull/40136/commits/4b90b64e70a5ab3701235f1be27fe0ec6aae037d

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx (hunk 3 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx]: Updated in https://github.com/woocommerce/woocommerce/pull/40136/commits/734032701872d77743de45a07add712e3a79fede

**Review by chihsuan — APPROVED** [review]
@moon0326 Looks good to me! 👍 

I've also pushed a few commits to fix the preview frame navigation and max-height issues. Also, added a scroll to the header/footer feature.

**Pull request merged**